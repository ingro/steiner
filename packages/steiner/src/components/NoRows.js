import React, { Component } from 'react';

import PropTypes from 'prop-types';

export default class NoRows extends Component {
    render() {
        const { errorMsg, isFetching, loadingMsg, noRowsMsg, itemsNumber } = this.props;

        const rows = itemsNumber || 0;

        if (isFetching && rows === 0) {
            return <div className="alert alert-info text-center">{loadingMsg}</div>;
        }

        if (errorMsg) {
            return <div className="alert alert-danger text-center">
                {errorMsg}
            </div>;
        }

        if (rows > 0) {
            return null;
        }

        return <div className="alert alert-warning text-center">{noRowsMsg}</div>;
    }
}

NoRows.propTypes = {
    errorMsg: PropTypes.string,
    isFetching: PropTypes.bool,
    itemsNumber: PropTypes.number,
    loadingMsg: PropTypes.string,
    noRowsMsg: PropTypes.string
};

NoRows.defaultProps = {
    loadingMsg: 'Loading...',
    noRowsMsg: 'No items to show'
};