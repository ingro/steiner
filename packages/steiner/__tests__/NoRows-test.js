import renderer from 'react-test-renderer';
import React from 'react';

import NoRows from '../src/components/NoRows';

test('NoRows renders correctly', () => {
    const tree = renderer.create(
        <NoRows />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});