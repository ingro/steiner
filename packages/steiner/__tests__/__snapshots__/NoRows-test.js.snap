exports[`test NoRows renders correctly 1`] = `
<div
  className="alert alert-warning text-center">
  No items to show
</div>
`;
